blobs/howto/python/blob-devguide-py/blob-devguide-blobs.py

```diff
@@ -0,0 +1,410 @@
+            print(f"Name: {blob['name']}, Tags: {blob['tags']}")
+    # </Snippet_list_blobs_flat_options> 
+
+    # <Snippet_list_blobs_hierarchical>
+    depth = 0
+    indent = "  "
+    def list_blobs_hierarchical(self, container_client: ContainerClient, prefix):
+        for blob in container_client.walk_blobs(name_starts_with=prefix, delimiter='/'):
+            if isinstance(blob, BlobPrefix):
+                # Indentation is only added to show nesting in the output
+                print(f"{self.indent * self.depth}{blob.name}")
+                self.depth += 1
+                self.list_blobs_hierarchical(container_client, prefix=blob.name)
+                self.depth -= 1
+            else:
+                print(f"{self.indent * self.depth}{blob.name}")
+    # </Snippet_list_blobs_hierarchical> 
+
+    # <Snippet_set_blob_tags>
+    def set_blob_tags(self, blob_service_client: BlobServiceClient, container_name):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        # Get any existing tags for the blob if they need to be preserved
+        tags = blob_client.get_blob_tags()
+
+        # Add or modify tags
+        updated_tags = {'Sealed': 'false', 'Content': 'image', 'Date': '2022-01-01'}
+        tags.update(updated_tags)
+
+        blob_client.set_blob_tags(tags)
+    # </Snippet_set_blob_tags>
+
+    # <Snippet_get_blob_tags>
+    def get_blob_tags(self, blob_service_client: BlobServiceClient, container_name):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        tags = blob_client.get_blob_tags()
+        print("Blob tags: ")
+        for k, v in tags.items():
+            print(k, v)
+    # </Snippet_get_blob_tags>
+
+    # <Snippet_clear_blob_tags>
+    def clear_blob_tags(self, blob_service_client: BlobServiceClient, container_name):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        # Pass in empty dict object to clear tags
+        tags = dict()
+        blob_client.set_blob_tags(tags)
+    # </Snippet_clear_blob_tags>
+
+    # <Snippet_find_blobs_by_tags>
+    def find_blobs_by_tags(self, blob_service_client: BlobServiceClient, container_name):
+        container_client = blob_service_client.get_container_client(container=container_name)
+
+        query = "\"Content\"='image'"
+        blob_list = container_client.find_blobs_by_tags(filter_expression=query)
+        
+        print("Blobs tagged as images")
+        for blob in blob_list:
+            print(blob.name)
+    # </Snippet_find_blobs_by_tags>
+
+    # <Snippet_copy_blob>
+    def copy_blob(self, blob_service_client: BlobServiceClient):
+        source_blob = blob_service_client.get_blob_client(container="source-container", blob="sample-blob.txt")
+
+        # Make sure the source blob exists before attempting to copy
+        if source_blob.exists:
```

Comment:
I think you want `.exists()` (function call instead of property)? I don't think we have a property on a `BlobClient` for its existence and simply creating the client won't make any network calls to determine its existence. The `exists()` function will actually make a HEAD network request to ensure the blob exists.

Reply:
Good catch, thanks!